### PR TITLE
Small enhancements for SF

### PR DESCRIPTION
**Pull request by t-b**
@MichaelHuth Please have a look how I implemented the two new features.

### CODE REVIEW

File: Packages/MIES/MIES_SweepFormula_Helpers.ipf [hunk 1]
```diff
@@ -432,23 +432,25 @@ Function/WAVE SFH_GetSweepsForFormula(string graph, WAVE range, WAVE/Z selectDat
...
 			for(j = 0; j < numEpochs; j += 1)
 				epIndex = epIndices[j]
-				range[0][j] = str2num(epochInfo[epIndex][EPOCH_COL_STARTTIME]) * ONE_TO_MILLI
-				range[1][j] = str2num(epochInfo[epIndex][EPOCH_COL_ENDTIME]) * ONE_TO_MILLI
+				adaptedRange[0][j] = str2num(epochInfo[epIndex][EPOCH_COL_STARTTIME]) * ONE_TO_MILLI
+				adaptedRange[1][j] = str2num(epochInfo[epIndex][EPOCH_COL_ENDTIME]) * ONE_TO_MILLI
 			endfor
 		else
-			Redimension/N=(-1, 1) range
+			Duplicate/FREE range, adaptedRange
```
Comment (MichaelHuth) [inline Packages/MIES/MIES_SweepFormula_Helpers.ipf:443]: Please add a note to the function description that the passed range should be treated as const.

File: Packages/tests/Basic/UTF_JSONWaveNotes.ipf [hunk 2]
```diff
@@ -184,3 +184,20 @@ static Function TestJSONWaveNoteCombinations()
...
 	CHECK_EQUAL_STR(str1, str)
 End
+
+static Function TestWaveNoteFromJSON()
+
+	variable jsonID = JSON_PARSE("{ \"abcd\" : [1, 2]}")
+	CHECK_GE_VAR(jsonID, 0)
+
+	Make/FREE wv
+	Note/K wv, "efgh\rJSON_BEGIN\r"
```
Comment (MichaelHuth) [inline Packages/tests/Basic/UTF_JSONWaveNotes.ipf:194]: Maybe we should have the WAVE_NOTE_JSON_SEPARATOR non-static then to be usable in tests.

**Review by MichaelHuth — CHANGES_REQUESTED** [review]
Looks good in general, just a few minor change requests.

File: Packages/MIES/MIES_SweepFormula.ipf [hunk 3]
```diff
@@ -1195,17 +1195,27 @@ Function [STRUCT RGBColor s] SF_GetTraceColor(string graph, string opStack, WAVE
 	return [s]
 End
 
-static Function [string traceName, variable traceCnt] SF_CreateTraceName(variable dataNum, STRUCT SF_PlotMetaData& plotMetaData, WAVE data)
+/// @brief Generate `numTraces` trace names for the given input
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_SweepFormula.ipf:1205]: Done

File: Packages/MIES/MIES_SweepFormula.ipf [hunk 4]
```diff
@@ -1663,6 +1663,20 @@ static Function SF_FormulaPlotter(string graph, string formula, [DFREF dfr, vari
...
+
+					WAVE/Z customMarkerAsFree = JWN_GetNumericWaveFromWaveNote(wvY, SF_META_MOD_MARKER)
+
+					if(WaveExists(customMarkerAsFree))
+						DFREF dfrWork = SFH_GetWorkingDF(graph)
+						for(i = 0; i < numTraces; i += 1)
+
+							wvName = UniqueWaveName(dfr, "customMarker_" + NameOfWave(wvY))
+							MoveWave customMarkerAsFree, dfrWork:$wvName
+							WAVE/SDFR=dfrWork customMarker = $wvName
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_SweepFormula.ipf:1683]: Good point. Done.

File: Packages/MIES/MIES_Constants.ipf [hunk 5]
```diff
@@ -1908,6 +1908,7 @@ StrConstant SF_META_XVALUES = "/XValues" // numeric wave
 StrConstant SF_META_XAXISLABEL = "/XAxisLabel" // string
 StrConstant SF_META_YAXISLABEL = "/YAxisLabel" // string
 StrConstant SF_META_OPSTACK = "/OperationStack" // string
+StrConstant SF_META_MOD_MARKER = "/Marker" // numeric wave
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_Constants.ipf:1912]: Done.

File: Packages/MIES/MIES_Constants.ipf [hunk 6]
```diff
@@ -1909,6 +1909,7 @@ StrConstant SF_META_XAXISLABEL = "/XAxisLabel" // string
 StrConstant SF_META_YAXISLABEL = "/YAxisLabel" // string
 StrConstant SF_META_OPSTACK = "/OperationStack" // string
 StrConstant SF_META_MOD_MARKER = "/Marker" // numeric wave
+StrConstant SF_META_SHOW_LEGEND = "/ShowLegend" // numeric, defaults to yes
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_Constants.ipf:1913]: Done.

File: Packages/MIES/MIES_SweepFormula_Helpers.ipf [hunk 7]
```diff
@@ -808,3 +809,17 @@ Function/S SFH_GetBrowserForFormulaGraph(string win)
 
 	return GetUserData(win, "", SFH_USER_DATA_BROWSER)
 End
+
+/// @brief Recreate the select data wave and range stored in the JSON wavenote from SFH_GetSweepsForFormula()
```
[the remark this answers is not in the document]
Comment (t-b) [reply Packages/MIES/MIES_SweepFormula_Helpers.ipf:830]: Done.

**Comment by t-b** [comment]
@MichaelHuth I've responded to the feedback.

**Pull request merged**